src/platforms/windows/windowsapplistprovider.cpp

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "windowsapplistprovider.h"
+#include "leakdetector.h"
+#include "logger.h"
+#include <QSettings>
+#include <QFile>
+#include <QDir>
+#include <QDirIterator>
+#include <QStandardPaths>
+#include <Windows.h>
+
+namespace {
+Logger logger(LOG_WINDOWS, "AppListProvider");
+}
+
+
+WindowsAppListProvider::WindowsAppListProvider(QObject* parent)
+    : AppListProvider(parent) {
+  MVPN_COUNT_CTOR(WindowsAppListProvider);
+}
+
+WindowsAppListProvider::~WindowsAppListProvider() {
+  MVPN_COUNT_DTOR(WindowsAppListProvider);
+}
+
+/**
+ * @brief Assembles an Application list through reading all links in common locations
+ */
+void WindowsAppListProvider::getApplicationList() {
+  QMap<QString, QString> appList;
+
+  readLinkFiles("C:\\ProgramData\\Microsoft\\Windows\\Start Menu\\Programs", &appList);
```

Comment:
instead of passing the appList as a pointer, what about if you pass it as a reference?
Or to make the code easier to read, you can return a list and append it to the appList.